### PR TITLE
[GEN-46] Feat: mixpanel tracking, node 18, react 18

### PR DESCRIPTION
**Pull request by rajatkantinandi**
- Mixpanel tracking
- Using .env & gitignore env file
- node 18, react 18 & latest version of mantine core v5
- Add isAddedFromCatalog in quiz object for tracking

Sample report:
https://eu.mixpanel.com/s/3F4lYL

### CODE REVIEW

File: src/config/local.ts [hunk 1]
```diff
@@ -4,8 +4,6 @@ const config: IConfig = {
   env: 'local',
   productName: 'Quiz Master',
   productUrl: 'http://localhost:5900',
-  backendUrl: 'http://localhost:3005',
```
Comment (govind007) [inline src/config/local.ts:7]: why did you remove this

Comment (rajatkantinandi) [reply src/config/local.ts:7]: Saved them in .env
